core/domain/base_model_validators.py

```diff
@@ -151,6 +160,11 @@ def __init__(
                 raise utils.ValidationError(
                     'The field \'%s\' should not contain '
                     'pseudonymous IDs' % field_name)
+        for model_id in filtered_model_ids:
+            if not user_services.is_user_id_valid(model_id):
```

Comment:
I'm not sure about adding this check here because I expect this class to just hold the data for fetching model details instead of this model also handing some validation. Is there any issue with ExternalModel validator validating model ids? [I'm fine with having this check, just wanted to check whether there's any issue with the existing structure?]

Reply:
We can probably shift it to _fetch_field_name_to_external_model_references method in the baseModelValidators before the point where the models for these ids are being fetched.

Reply:
I think the useful thing about this is that we can have different fetchers for different kinds of external models and the logic is clearly separated into classes.

Reply:
> We can probably shift it to _fetch_field_name_to_external_model_references method in the baseModelValidators before the point where the models for these ids are being fetched.

I agree with @rohitkatlaa here i.e, this function should not validate anything related to external models instead it should just prepare the details and we should try to fetch the external models with id (Whatever it is), and if models not found then raise an error. @vojtechjelinek wdyt?

Reply:
As I said I think that validation like this can be placed in the class directly (similar to how we have `validate` in domain objects), also now it doesn`t call a `user_services` since `is_user_id_valid` was moved to `utils`.

Is there any other reason why to move the code away from this place? I do not want to waste time working on it without any strong reason.

Reply:
> Is there any other reason why to move the code away from this place?

The major reason was to avoid doing model field validations in *FetecherDetails class and do the validations in and function/class which specifies it handles/does validation. I took a pass on `is_user_id_valid`and it seems like a basic pattern check (the name of the function is a bit misnomer) and I think it's fine to have this basic pattern check [but in general, I think we should avoid doing model filed validations in the FetecherDetailsClass.]